Pcap++/src/PfRingDeviceList.cpp

```diff
@@ -4,14 +4,53 @@
 
 #define LOG_MODULE PcapLogModulePfRingDevice
 
+#include <cstdio>
 #include "PfRingDeviceList.h"
 #include "SystemUtils.h"
+#include "DeviceUtils.h"
 #include "Logger.h"
 #include "pcap.h"
 #include "pfring.h"
 
 namespace pcpp
 {
+	/// @cond PCPP_INTERNAL
+
+	namespace
+	{
+		/**
+		 * @class PfRingCloseDeleter
+		 * A deleter that cleans up a pfring structure by calling pfring_close.
+		 */
+		struct PfRingCloseDeleter
+		{
+			void operator()(pfring* ptr) const { pfring_close(ptr); }
+		};
+
+		/**
+		 * Reads the ring version of a PF_RING handle.
+		 * @param[in] ring A PF_RING handle.
+		 * @return A string representation of the ring version.
+		 */
+		std::string readPfRingVersion(pfring* ring)
+		{
+			uint32_t version;
+			if (pfring_version(ring, &version) < 0)
+			{
+				throw std::runtime_error("Couldn't retrieve PF_RING version, pfring_version returned an error");
+			}
+
+			char versionAsString[25];
+			std::snprintf(versionAsString, 25, "PF_RING v.%u.%u.%u\n",
+				(version & 0xFFFF0000) >> 16,
+				(version & 0x0000FF00) >> 8,
+				 version & 0x000000FF);
+
+			return std::string(versionAsString);
+		}
+	}
```

Comment:
I'm not sure this needs to be in its own namespace. Why not just put it in the cpp file?

Reply:
I guess, it can be. I wasn't exactly sure on, if `Foo::(unnamed namespace)` also had internal linkage, but reading more about it, they do.

Reply:
does it really matter? It's anyway not exposed to the outside 🤔

Reply:
I find it more clear having `internal to this file only` functions wrapped in an unnamed namespace, so they can't be accidentally exported. Other than that and the internal linkage, not really sure.

Reply:
ok sure 👍